crates/relayer/src/chain/ckb4ibc/utils.rs

```diff
@@ -115,12 +119,18 @@ pub fn get_connection_search_key(
         client_id = Some(config.lc_client_id(client_type)?.to_string());
     }
     let script = get_connection_lock_script(config, client_id)?;
+    let script_search_mode = if client_type.is_some() {
+        Some(ScriptSearchMode::Exact)
+    } else {
+        Some(ScriptSearchMode::Prefix)
```

Comment:
If we use `ScriptSearchMode::Prefix`, we must check the script args again after we get the result. Otherwise we may receive a fake cell.(attacker append junk data after the script args to match our search key)

Reply:
makes sense, I'll apply a filter in the result of prefix search